### PR TITLE
Get rid of the unnecessary AccountDao.constructAccount method.

### PR DESCRIPTION
**Pull request by alxdarksage**
The account classes are complicated enough without this unnecessary call into the account DAO.

### CODE REVIEW

File: test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java [hunk 1]
```diff
@@ -320,26 +327,33 @@ public void createParticipant() {
...
         verify(accountWorkflowService).sendEmailVerificationToken(STUDY, ID, EMAIL);
         
         Account account = accountCaptor.getValue();
+        assertEquals(STUDY.getIdentifier(), account.getStudyId());
+        assertEquals(EMAIL, account.getEmail());
+        assertEquals(PHONE, account.getPhone());
+        assertEquals(EXTERNAL_ID, account.getExternalId());
+        assertNotNull(account.getPasswordHash());
+        assertEquals(PasswordAlgorithm.DEFAULT_PASSWORD_ALGORITHM, account.getPasswordAlgorithm());
+        assertNotEquals(PASSWORD, account.getPasswordHash());
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java:354]: We should also check for the following attributes: id, phoneVerified, healthCode.

File: app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java [hunk 2]
```diff
@@ -270,31 +270,6 @@ private boolean updateReauthToken(Study study, Account hibernateAccount) {
         return true;
     }
     
-    /** {@inheritDoc} */
-    @Override
-    public Account constructAccount(Study study, String email, Phone phone, String externalId, String password) {
-        // Set basic params from inputs.
-        Account account = Account.create();
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java:277]: There should be an update to HibernateAccountDaoTest.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java (hunk 1 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java:354]: Added

File: app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java (hunk 2 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java:277]: Yes... this PR doesn't even compile. Removed two test methods referencing constructAccount(), ran unit tests.

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**